drivers/ethernet/eth_sam_gmac_priv.h

```diff
@@ -10,6 +10,8 @@
 #ifndef ZEPHYR_DRIVERS_ETHERNET_ETH_SAM_GMAC_PRIV_H_
 #define ZEPHYR_DRIVERS_ETHERNET_ETH_SAM_GMAC_PRIV_H_
 
+#define DT_DRV_COMPAT atmel_sam_gmac
```

Comment:
Move this from the header into the top of the driver source file. (before any includes).

Reply:
Done.

FYI, `eth_sam_gmac_priv.h` uses `DT_INST_PROP` macro, and we will be seeing very strange behaviours if this header gets included by any other file (though, since this is a "private header," it shouldn't really matter).